lib/Bio/EnsEMBL/DataCheck/Checks/CigarCheck.pm

```diff
@@ -0,0 +1,82 @@
+=head1 LICENSE
+
+Copyright [2018-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::CigarCheck;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'CigarCheck',
+  DESCRIPTION    => 'The cigar_line must not have negative numbers or zeros in it',
```

Comment:
There is no `GROUPS` parameter set here. It's not mandatory for a datacheck to belong to a group, but it is expected, and it makes the datachecks much more useful.

Groups can be named anything, but for fundamental tests, which no database should fail (i.e. the set which we will run at handover) the convention is to use the type of database, so 'compara'. In addition, you can specify as many other groups as you like, in order to logically group datachecks. I'm not sure if there's a good group in this case, but the other datacheck in this PR might have something like 'pairwise_alignments'.

You can browse how other teams have grouped things: http://production-services.ensembl.org/app/datachecks-list

As well as keeping things neat, groups can help a lot when embedding datachecks in pipelines. For example, after running the ProteinFeatures pipeline, we run the 'protein_features' group: https://github.com/Ensembl/ensembl-production/blob/release/98/modules/Bio/EnsEMBL/Production/Pipeline/PipeConfig/ProteinFeatures_conf.pm#L909

```suggestion
  DESCRIPTION    => 'The cigar_line must not have negative numbers or zeros in it',
  GROUPS        => ['compara'],
```

Reply:
Let's establish the convention, then.
1. `compara` will be listed in every DC. It's the complete list of all DCs that apply to a Compara databae
2. Other groups will be prefixed by `compara_` to make them easier to list and identify, so `compara_pairwise_alignments`, not just `pairwise_alignments`
3. We use the plural form, e.g. `compara_pairwise_alignments` not `compara_pairwise_alignment`

Does it make sense ?

@CristiGuijarro : `compara_pairwise_alignments` should be added to _CigarCheck_ as well because it applies to the pairwise-alignments databases too (amongst other things). The same way that _TagCoverageStats_ applies to other things than pairwise-alignments (multiple alignments, syntenies)

NB: We'll probably add more groups to these two DCs, e.g. `compara_syntenies`, `compara_gene_trees`, etc, when we add other DCs and start defining extra groups

Reply:
Sounds sensible to me.